stdlib/public/Distributed/DistributedActorSystem.swift

```diff
@@ -849,6 +849,27 @@ public struct RemoteCallArgument<Value> {
 /// /// performs the actual distributed (local) instance method invocation.
 /// mutating func decodeNextArgument<Argument: SerializationRequirement>() throws -> Argument
 /// ```
+///
+/// ### Decoding DistributedActor arguments using Codable
+/// When using the actor system's ``ActorID`` is ``Codable``, every distributed actor using that system
+/// is also implicitly ``Codable`` (see ``DistributedActorSystem``). Such distributed actors are encoded
+/// as their ``ActorID`` stored in a ``Encoder/singleValueContainer``. Since ``Codable`` is being used
+/// by this such system, it means that the ``decodeNextArgument`` method will be using ``Decoder`` to
+/// decode the incoming values, some of which may be distributed actors.
+///
+/// In order for a distributed actor's ``Decodable/init(from:)`` to be able to return an existing instance,
+/// i.e. when the being-decoded ``ActorID`` actually references a locally hosted distributed actor, that the
+/// current actor system can return from `resolve(id:as:)`, an actor system must be provided to the ``Decoder``.
+/// This is done by setting the actor system the decoding is performed for, on the decoder's `userInfo`, as follows:
+///
+/// ```
+/// mutating func decodeNextArgument<Argument: SerializationRequirement>() throws -> Argument {
+///   let argumentData: Data = /// ...
+///   // ...
+///   decoder.userInfo[.actorSystemKey] = self.actorSystem
+///   return try Argument.decode(
+/// }
+// ```
```

Comment:
Some suggested tweaks based on my understanding of things. Hopefully I got the details right.

```suggestion
/// ### Decoding DistributedActor arguments using Codable
/// When using an actor system where ``ActorID`` is ``Codable``, every distributed actor using that system
/// is also implicitly ``Codable`` (see ``DistributedActorSystem``). Such distributed actors are encoded
/// as their ``ActorID`` stored in an ``Encoder/singleValueContainer``. When ``Codable`` is being used
/// by such a system, the ``decodeNextArgument`` method will be using ``Decoder`` to
/// decode the incoming values, which may themselves be distributed actors.
///
/// An actor system must be provided to the ``Decoder`` in order for a distributed actor's ``Decodable/init(from:)`` /// to be able to return the instance of the actor. Specifically, when the ``ActorID`` references a locally hosted 
/// distributed actor, the actor system's `resolve(id:as:)` method will be used to find the instance of the actor.
/// Thus, you must set the actor system the decoding is performed for, on the decoder's `userInfo`, as follows:
///
/// ```
/// mutating func decodeNextArgument<Argument: SerializationRequirement>() throws -> Argument {
///   let argumentData: Data = /// ...
///   // ...
///   decoder.userInfo[.actorSystemKey] = self.actorSystem
///   return try Argument.decode(
/// }
// ```
```

Reply:
Thanks, also noticed a logical typo thanks to your rewording so fixing now all together